### PR TITLE
Fix possible deadlock in urgent state of report uploader

### PR DESCRIPTION
**Pull request by wowlocal**
Added forgotten semaphore_signal calls at early return

### CODE REVIEW

**Comment by ncooke3** [comment]
Hi @wowlocal, thanks for opening a PR and signing the CLA!

File: Crashlytics/UnitTests/FIRCLSReportUploaderTests.m [hunk 1]
```diff
@@ -114,17 +115,46 @@ - (void)testUrgentUploadPackagedReportWithPath {
   [self runUploadPackagedReportWithUrgency:YES];
 }
 
-- (void)testUploadPackagedReportWithoutDataCollectionToken {
-  [self setUpForUpload];
+- (void)testUrgentWaitUntillUpload {
```
Comment (ncooke3) [inline Crashlytics/UnitTests/FIRCLSReportUploaderTests.m:118]: Curious, do these added tests deadlock without the semaphore signaling?

Comment (wowlocal) [reply Crashlytics/UnitTests/FIRCLSReportUploaderTests.m:118]: Yep) tests will fail with the timeout error

**Review by samedson — APPROVED** [review]
This is a great fix - thank you for contributing! LGTM

**Comment by elenadoty** [comment]
This is great, thanks @wowlocal !

**Review by elenadoty — APPROVED** [review]

**Comment by elenadoty** [comment]
Hi @wowlocal - I'm working on Google Open Source Peer Bonus nominations and would like to recognize your contribution to improving Crashlytics. I couldn't find an email address for you, so if you're interested please email it to elenadoty at google.com and I'll make the nomination.

**Pull request merged**